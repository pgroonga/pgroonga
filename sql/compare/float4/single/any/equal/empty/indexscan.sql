CREATE TABLE ids (
  id real
);

INSERT INTO ids VALUES (1.1);
INSERT INTO ids VALUES (2.1);
INSERT INTO ids VALUES (3.1);

CREATE INDEX pgroonga_index ON ids USING pgroonga (id pgroonga_float4_ops);

SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;

EXPLAIN (COSTS OFF)
SELECT id
  FROM ids
 WHERE id = ANY(ARRAY[]::real[])
 ORDER BY id ASC;

SELECT id
  FROM ids
 WHERE id = ANY(ARRAY[]::real[])
 ORDER BY id ASC;

DROP TABLE ids;
