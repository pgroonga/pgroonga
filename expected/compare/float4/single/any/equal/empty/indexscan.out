CREATE TABLE ids (
  id real
);
INSERT INTO ids VALUES (1.1);
INSERT INTO ids VALUES (2.1);
INSERT INTO ids VALUES (3.1);
CREATE INDEX pgroonga_index ON ids USING pgroonga (id pgroonga_float4_ops);
SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;
EXPLAIN (COSTS OFF)
SELECT id
  FROM ids
 WHERE id = ANY(ARRAY[]::real[])
 ORDER BY id ASC;
               QUERY PLAN                
-----------------------------------------
 Index Scan using pgroonga_index on ids
   Index Cond: (id = ANY ('{}'::real[]))
(2 rows)

SELECT id
  FROM ids
 WHERE id = ANY(ARRAY[]::real[])
 ORDER BY id ASC;
 id 
----
(0 rows)

DROP TABLE ids;
